lib/optimizely.rb

```diff
@@ -470,8 +428,49 @@ def get_feature_variable_integer(feature_flag_key, variable_key, user_id, attrib
       variable_value
     end
 
+    def is_valid
+      config = project_config
+      config.is_a?(Optimizely::ProjectConfig)
+    end
+
     private
 
+    def get_variation_with_config(experiment_key, user_id, attributes, config)
+      # Gets variation where visitor will be bucketed.
+      #
+      # experiment_key - Experiment for which visitor variation needs to be determined.
+      # user_id - String ID for user.
+      # attributes - Hash representing user attributes.
+      # config - Instance of DatfileProjectConfig
+      #
+      # Returns [variation key] where visitor will be bucketed.
+      # Returns [nil] if experiment is not Running, if user is not in experiment, or if datafile is invalid.
+      experiment = config.get_experiment_from_key(experiment_key)
+      return nil if experiment.nil?
+
+      unless user_inputs_valid?(attributes)
+        @logger.log(Logger::INFO, "Not activating user '#{user_id}.")
```

Comment:
This is used by both `get_variation` and `activate`

Reply:
Just removed it; Already being used in `user_inputs_valid?`.